### ISSUE TITLE
[terra-select] Update placeholders and allowClear

### ISSUE THREAD

**Issue by neilpfeiffer** [cerner/terra-core#3083]
# Feature Request

## Description
As part of removing placeholders in form inputs for better accessibility, the Form Select inputs will also have a higher level of Cerner UX Design Standard requirements to provide a common and consistent placeholder pattern that is no longer editable by consumers. 

Similar to Native Select, `allowClear` now becomes a default feature included with SingleSelect, and `isFilterStyle` is added for cases where a value must always be selected and not cleared, commonly found in filters and toolbars.

Related to 
[[terra-form-input] Insufficient color-contrast for placeholder #2992](https://github.com/cerner/terra-core/issues/2992)

**Comment by neilpfeiffer** [comment]
The alteration direction below is copied from https://github.com/cerner/terra-core/issues/2992#issuecomment-651135365

---

Alteration directions for:

- terra-form-select
- terra-form-select/lib/SelectField
- terra-form-select/lib/SingleSelect
- terra-form-select/lib/SingleSelectField
- terra-form-select/lib/Combobox
- terra-form-select/lib/ComboboxField
- terra-form-select/lib/MultiSelect
- terra-form-select/lib/MultiSelectField
- terra-form-select/lib/TagSelect
- terra-form-select/lib/TagSelectField
- terra-form-select/lib/SearchSelect
- terra-form-select/lib/SearchSelectField

---
- [ ] All forms of select: remove the `placeholder` prop that allows consumers to customize the placeholder text
- [ ] All forms of select: the default text given to the component as the placeholder is `- Select -`
- [ ] For SingleSelect, remove the `allowClear` prop, making the functionality the default behavior.
- [ ] For SingleSelect, add a new prop `isFilterStyle` to match native-select that removes the ability to have no value selected, showing a placeholder, and ability to clear an already selected value.
- [ ] Theming changes:
**Default theme:**
placeholder color:  `rgba(21, 33, 41, 0.62)`
placeholder font-style: `italic`
disabled placeholder color:`rgba(21, 33, 41, 0.62)`
**Fusion theme:**
placeholder color:  _TODO:ux-check-if-needs-to-be-updated_
placeholder font-style: `normal`
disabled placeholder color: _TODO:ux-check-if-needs-to-be-updated_
**Lowlight theme:**
placeholder color:  `#6f7477`
placeholder font-style: `italic`
disabled placeholder color: `#6f7477`
- [ ] Add major version Upgrade Guide that includes these comments:
_TODO_ add the language needed for the accessibility and human factors guidance switching from consumers providing placeholders to providing more detailed help text with Field compositions.
